android/app/src/main/java/updated/mysterium/vpn/ui/manual/connect/filter/FilterActivity.kt

```diff
@@ -0,0 +1,178 @@
+import androidx.recyclerview.widget.LinearLayoutManager
+import network.mysterium.vpn.R
+import network.mysterium.vpn.databinding.FragmentNodeListBinding
+import network.mysterium.vpn.databinding.ToolbarBaseConnectBinding
+import org.koin.android.ext.android.inject
+import updated.mysterium.vpn.model.filter.NodePrice
+import updated.mysterium.vpn.model.filter.NodeQuality
+import updated.mysterium.vpn.model.filter.NodeType
+import updated.mysterium.vpn.model.manual.connect.CountryNodesModel
+import updated.mysterium.vpn.model.manual.connect.ProposalModel
+import updated.mysterium.vpn.ui.manual.connect.BaseConnectActivity
+
+class FilterActivity : BaseConnectActivity() {
+
+    companion object {
+
+        var countryNodesModel: CountryNodesModel? = null
+    }
+
+    private lateinit var binding: FragmentNodeListBinding
+    private val nodeListAdapter = FilterAdapter()
+    private val viewModel: FilterViewModel by inject()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = FragmentNodeListBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        initProposalListRecycler()
+        bindsActions()
+        subscribeViewModel()
+        getNodesList()
+    }
+
+    override fun reUseToolbar(toolbarBinding: ToolbarBaseConnectBinding) {
+        changeRightIcon(R.drawable.icon_search)
+        if (toolbarBinding.root.parent != null) {
+            (toolbarBinding.root.parent as ViewGroup).removeView(toolbarBinding.root)
+        }
+        binding.manualConnectToolbar.addView(toolbarBinding.root)
+    }
+
+    private fun getNodesList() {
+        countryNodesModel?.let {
+            if (it.countryName.isEmpty()) {
+                binding.nodesTitle.text = getString(R.string.manual_connect_all_countries)
+                nodeListAdapter.isCountryNamedMode = true
+            } else {
+                binding.nodesTitle.text = it.countryName
+            }
+            viewModel.applyInitialFilter(it)
+        }
+    }
+
+    private fun subscribeViewModel() {
+        viewModel.proposalsList.observe(
+            this,
+            { proposals -> showNodeList(proposals) }
+        )
+    }
+
+    private fun bindsActions() {
+        binding.filters.typeCardView.setOnClickListener {
+            viewModel.onNodeTypeClicked().observe(
+                this,
+                { result -> result.onSuccess { changeNodeTypeView(it) } }
+            )
+        }
+        binding.filters.priceCardView.setOnClickListener {
+            viewModel.onNodePriceClicked().observe(
+                this,
+                { result -> result.onSuccess { changeNodePriceView(it) } }
+            )
+        }
+        binding.filters.qualityCardView.setOnClickListener {
+            viewModel.onNodeQualityClicked().observe(
+                this,
+                { result -> result.onSuccess { changeNodeQualityView(it) } }
+            )
+        }
+    }
```

Comment:
Body better be written on separate lines

Reply:
Fix